src/docs/slack-tracking.md

```diff
@@ -0,0 +1,399 @@
+
+Again, it's this interesting "blindness" of JavaScript about what will happen next. I think of the Map (hiddden class) displayed above as "the" Map for a `Peak` object, but V8 doesn't have that luxury. The Map above is only a leaf in a chain of descendents from the **initial map** at the start of the function.
+
+It's the construction counter in the **initial map** that we use to control slack tracking.
+
+How to find the **initial map**? Well, the function `Peak()` has a pointer to it:
+
+```
+> %DebugPrint(Peak);
+d8> %DebugPrint(Peak)
+DebugPrint: 0x31c12561: [Function] in OldSpace
+ - map: 0x2a2821f5 <Map(HOLEY_ELEMENTS)> [FastProperties]
+ - prototype: 0x31c034b5 <JSFunction (sfi = 0x36108421)>
+ - elements: 0x28c821a1 <FixedArray[0]> [HOLEY_ELEMENTS]
+ - function prototype: 0x37449c89 <Object map = 0x2a287335>
+ - initial_map: 0x46f07295 <Map(HOLEY_ELEMENTS)>        // Here's the initial map.
+ - shared_info: 0x31c12495 <SharedFunctionInfo Peak>
+ - name: 0x31c12405 <String[4]: #Peak>
+...
+... (etc)
+...
+d8> %DebugPrintPtr(0x46f07295)                          // %DebugPrintPtr allows you to print the initial map
+DebugPrint: 0x46f07295: [Map]
+ - type: JS_OBJECT_TYPE
+ - instance size: 52
+ - inobject properties: 10
+ - elements kind: HOLEY_ELEMENTS
+ - unused property fields: 10
+ - enum length: invalid
+ - back pointer: 0x28c02329 <undefined>
+ - prototype_validity cell: 0x47f0232d <Cell value= 1>
+ - instance descriptors (own) #0: 0x28c02135 <DescriptorArray[0]>
+ - transitions #1: 0x46f0735d <Map(HOLEY_ELEMENTS)>
+     0x28c046f9: [String] in ReadOnlySpace: #name: (transition to (const data field, attrs: [WEC]) @ Any)
+         -> 0x46f0735d <Map(HOLEY_ELEMENTS)>
+ - prototype: 0x5cc09c7d <Object map = 0x46f07335>
+ - constructor: 0x21e92561 <JSFunction Peak (sfi = 0x21e92495)>
+ - dependent code: 0x28c0212d <Other heap object (WEAK_FIXED_ARRAY_TYPE)>
+ - construction counter: 5
+```
+
+See how the construction counter is decremented to 5. The initial map, also called the root map, has no back pointer. If you'd like to find the initial map from the two property map we showed above, you can follow it's back pointer with the help of `%DebugPrintPtr()` until you reach a map with `undefined` in the back pointer slot. That will be this map above.
+
+A tree grows from the root map, with a branch for each property added from that point. The current structure looks like this:
+
+<figure>
+  <img src="/_img/docs/slack-tracking/root-map-1.svg" loading="lazy">
+  <figcaption>
+  <i>(X, Y, Z) means (instance size, number of in-object properties, number of unused properties)</i>
+  </figcaption>
+</figure>
+
+These transitions based on property names are how the "blind mole" of JavaScript builds it's hidden classes behind you. This root map is also stored in the function 'Peak', so when it's used as a constructor, that map can be used to set up the `this` object.
```

Comment:
nit: mode

Reply:
Where would that be? I want the image of a mole, the essentially blind creature that digs in the dirt, finding it's way carefully forward. "Blind mode" would then suggest that there is some other mode of JavaScript, but that would probably be wasm.